fdb-record-layer-core/src/test/java/com/apple/foundationdb/record/provider/foundationdb/RecordTypeKeyTest.java

```diff
@@ -697,5 +698,96 @@ private void saveManyRecords(@Nonnull RecordMetaDataHook hook, int count1, int c
             context.commit();
         }
     }
+    
+    @Test
+    public void testOnlineIndexBuilderMaxWrite() throws Exception {
+        try (FDBRecordContext context = openContext()) {
+            uncheckedOpenSimpleRecordStore(context, BASIC_HOOK);
+            recordStore.checkVersion(null, FDBRecordStoreBase.StoreExistenceCheck.ERROR_IF_EXISTS).join();
+            context.commit();
+        }
+
+        saveManyRecords(BASIC_HOOK, 250, 250);
 
+        try (FDBRecordContext context = openContext()) {
+            uncheckedOpenSimpleRecordStore(context, BASIC_HOOK);
+            recordStore.checkVersion(null, FDBRecordStoreBase.StoreExistenceCheck.ERROR_IF_NOT_EXISTS).join();
+
+            assertEquals(250, recordStore.getSnapshotRecordCountForRecordType("MySimpleRecord").join().intValue());
+            assertEquals(250, recordStore.getSnapshotRecordCountForRecordType("MyOtherRecord").join().intValue());
+        }
+
+        RecordMetaDataHook hook = metaData -> {
+            BASIC_HOOK.apply(metaData);
+            metaData.addIndex("MySimpleRecord", "newIndex", "num_value_2");
+        };
+
+        try (FDBRecordContext context = openContext()) {
+            uncheckedOpenSimpleRecordStore(context, hook);
+            recordStore.checkVersion(null, FDBRecordStoreBase.StoreExistenceCheck.ERROR_IF_NOT_EXISTS).join();
+
+            assertTrue(recordStore.isIndexWriteOnly("newIndex"));
+
+            timer.reset();
+
+            // Build in this transaction.
+            try (OnlineIndexer indexBuilder =
```

Comment:
I think some old code still calls is `indexBuilder` (`OnlineIndexer` used to be called `OnlineIndexBuilder`) but it'd better to be called `indexer` because it's weird to have a "Builder" for a "XXXBuilder".